component/loki/source/azureeventhubs/azureeventhubs.go

```diff
@@ -0,0 +1,210 @@
+package azureeventhubs
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/Shopify/sarama"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	flow_relabel "github.com/grafana/agent/component/common/relabel"
+	"github.com/grafana/agent/component/loki/source/azureeventhubs/internal/parser"
+	kt "github.com/grafana/agent/component/loki/source/internal/kafkatarget"
+	"github.com/grafana/dskit/flagext"
+
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "loki.source.azure_event_hubs",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// Arguments holds values which are used to configure the loki.source.azure_event_hubs component.
+type Arguments struct {
+	FullyQualifiedNamespace string   `river:"fully_qualified_namespace,attr"`
+	EventHubs               []string `river:"event_hubs,attr"`
+
+	Authentication AzureEventHubsAuthentication `river:"authentication,block"`
+
+	GroupID                string             `river:"group_id,attr,optional"`
+	UseIncomingTimestamp   bool               `river:"use_incoming_timestamp,attr,optional"`
+	DisallowCustomMessages bool               `river:"disallow_custom_messages,attr,optional"`
+	RelabelRules           flow_relabel.Rules `river:"relabel_rules,attr,optional"`
+	Labels                 map[string]string  `river:"labels,attr,optional"`
+	Assignor               string             `river:"assignor,attr,optional"`
+
+	ForwardTo []loki.LogsReceiver `river:"forward_to,attr"`
+}
+
+// AzureEventHubsAuthentication describe the configuration for authentication with Azure Event Hub
+type AzureEventHubsAuthentication struct {
+	Mechanism        string   `river:"mechanism,attr"`
+	Scopes           []string `river:"scopes,attr,optional"`
+	ConnectionString string   `river:"connection_string,attr,optional"`
+}
+
+func getDefault() Arguments {
+	return Arguments{
+		GroupID:  "loki.source.azure_event_hubs",
+		Labels:   map[string]string{"job": "loki.source.azure_event_hubs"},
+		Assignor: "range",
+	}
+}
+
+// UnmarshalRiver implements river.Unmarshaler.
+func (a *Arguments) UnmarshalRiver(f func(interface{}) error) error {
+	*a = getDefault()
+	type arguments Arguments
+	return f((*arguments)(a))
```

Comment:
nit: We could validate the valid `assignor` values here, or at least put a TODO comment for this.